CoViewingExample/AudioDevices/ExampleAVPlayerAudioDevice.m

```diff
@@ -0,0 +1,1132 @@
+    /*
+     * Interruption handling is different on iOS 9.x. If your application becomes interrupted while it is in the
+     * background then you will not get a corresponding notification when the interruption ends. We workaround this
+     * by handling UIApplicationDidBecomeActiveNotification and treating it as an interruption end.
+     */
+    if (![[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){10, 0, 0}]) {
+        [center addObserver:self selector:@selector(handleApplicationDidBecomeActive:) name:UIApplicationDidBecomeActiveNotification object:nil];
+    }
+
+    [center addObserver:self selector:@selector(handleRouteChange:) name:AVAudioSessionRouteChangeNotification object:nil];
+    [center addObserver:self selector:@selector(handleMediaServiceLost:) name:AVAudioSessionMediaServicesWereLostNotification object:nil];
+    [center addObserver:self selector:@selector(handleMediaServiceRestored:) name:AVAudioSessionMediaServicesWereResetNotification object:nil];
+}
+
+- (void)handleAudioInterruption:(NSNotification *)notification {
+    AVAudioSessionInterruptionType type = [notification.userInfo[AVAudioSessionInterruptionTypeKey] unsignedIntegerValue];
+
+    @synchronized(self) {
+        // TODO: Multiple contexts.
+        // If the worker block is executed, then context is guaranteed to be valid.
+        TVIAudioDeviceContext context = self.renderingContext ? self.renderingContext->deviceContext : NULL;
+        if (context) {
+            TVIAudioDeviceExecuteWorkerBlock(context, ^{
+                if (type == AVAudioSessionInterruptionTypeBegan) {
+                    NSLog(@"Interruption began.");
+                    self.interrupted = YES;
+                    [self stopAudioUnit];
+                    TVIAudioSessionDeactivated(context);
+                } else {
+                    NSLog(@"Interruption ended.");
+                    self.interrupted = NO;
+                    if ([self startAudioUnit]) {
+                        TVIAudioSessionActivated(context);
+                    }
+                }
+            });
+        }
+    }
+}
+
+- (void)handleApplicationDidBecomeActive:(NSNotification *)notification {
+    @synchronized(self) {
+        // If the worker block is executed, then context is guaranteed to be valid.
+        TVIAudioDeviceContext context = self.renderingContext ? self.renderingContext->deviceContext : NULL;
+        if (context) {
+            TVIAudioDeviceExecuteWorkerBlock(context, ^{
+                if (self.isInterrupted) {
+                    NSLog(@"Synthesizing an interruption ended event for iOS 9.x devices.");
+                    self.interrupted = NO;
+                    if ([self startAudioUnit]) {
+                        TVIAudioSessionActivated(context);
+                    }
+                }
+            });
+        }
+    }
+}
+
+- (void)handleRouteChange:(NSNotification *)notification {
```

Comment:
did you get chance to test this?

Reply:
Not yet, though I do see route change events being fired during normal operation (That don't restart).